### PR TITLE
Simplifies Dockerfile

### PR DESCRIPTION
**Pull request by lavafroth**
- Opts out of .NET CLI telemetry
- Removes redundant Dockerfile commands (like copying pyproject.toml)
- Does not use sudo everywhere because we are already root inside the container
- Downloads Microsoft's Debian prod package list only once
- Updates and installs necessary packages through apt-get in a single run

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,44 +20,37 @@ LABEL maintainer="bc-security"
...
+    apt-transport-https \
+    xclip \
+    zip \
+    powershell \
+    dotnet-sdk-6.0 \
     && rm -rf /var/lib/apt/lists/*
 
 WORKDIR /empire
 
-COPY pyproject.toml /empire
```
Comment (vinnybod) [inline Dockerfile:53]: I think what would be most beneficial from a caching perspective would be something like:

```
COPY pyproject.toml poetry.lock /empire/

RUN pip install poetry \
    --disable-pip-version-check && \
    poetry config virtualenvs.create false && \
    poetry install --no-root

COPY . /empire
```

This way the dependency layer is cached even when we have code changes (without dependency changes). This was my original intent with the Dockerfile, but obv did not do it properly. What do you think?

Comment (lavafroth) [reply Dockerfile:53]: Sure thing

**Comment by vinnybod** [comment]
Can you explain the addition of `libicu-dev`? I want to be intentional about all the packages we are adding to the  container and I think issue https://github.com/BC-SECURITY/Empire/issues/609 was already solved by pr https://github.com/BC-SECURITY/Empire/pull/616 ?

**Comment by lavafroth** [comment]
The csharpserver build fails and we can notice
```
Process terminated. Couldn't find a valid ICU package installed on the system.
```

**Review by vinnybod — APPROVED** [review]
looks fine to me. I'll merge after I test and verify.

**Comment by vinnybod** [comment]
Can you merge the dev branch into your branch? That would allow me to run the github CI.

**Comment by lavafroth** [comment]
Merged.

**Pull request merged**